Author: José Manuel Santamaría Lema <panfaust@gmail.com>
Forwarded: no
Description: This patch makes muon-installer list properly unofficial packages'
 origins. Probably this should be fixed in libqapt instead.
---
 installer/ApplicationWindow.cpp |   12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

--- a/installer/ApplicationWindow.cpp
+++ b/installer/ApplicationWindow.cpp
@@ -253,7 +253,11 @@ void ApplicationWindow::populateViews()
         QString originLabel = m_backend->originLabel(originName);
         QStandardItem *viewItem = new QStandardItem;
         viewItem->setEditable(false);
-        viewItem->setText(originLabel);
+        if (originLabel.isEmpty()) {
+            viewItem->setText(originName);
+        } else {
+            viewItem->setText(originLabel);
+        }
         viewItem->setData(originName, OriginFilterRole);
         viewItem->setData(AppView, ViewTypeRole);
 
@@ -318,7 +322,11 @@ void ApplicationWindow::populateViews()
         QString originLabel = m_backend->originLabel(originName);
         QStandardItem *viewItem = new QStandardItem;
         viewItem->setEditable(false);
-        viewItem->setText(originLabel);
+        if (originLabel.isEmpty()) {
+            viewItem->setText(originName);
+        } else {
+            viewItem->setText(originLabel);
+        }
         viewItem->setData(QApt::Package::Installed, StateFilterRole);
         viewItem->setData(originName, OriginFilterRole);
         viewItem->setData(AppView, ViewTypeRole);
